Classes/Domain/Repository/HaikuRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\ConnectionPool;
+
+class HaikuRepository
+{
+    public function __construct(
```

Comment:
Maybe:

```
private readonly Connection $connection;

public function __construct(ConnectionPool $connectionPool)
{
    $this->connection = $connectionPool->getConnectionForTable('tx_examples_haiku');
}
```

This way you don't have to call everytime the getConnectionForTable in the methods.

Reply:
nice solution